praetorian_cli/sdk/entities/aegis.py

```
@@ -0,0 +1,437 @@
+        """
+        try:
+            agents_data = self.api.get('/aegis/agent')
+            
+            # Return Agent objects
+            agents = []
+            for agent_data in agents_data:
+                agent = Agent.from_dict(agent_data)
+                agents.append(agent)
+            return agents, None
+        except Exception as e:
+            raise Exception(f"Failed to list Aegis agents: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)